lib/ansible/modules/cloud/amazon/s3_lifecycle.py

```diff
@@ -467,6 +465,18 @@ def main():
     transition_date = module.params.get("transition_date")
     state = module.params.get("state")
 
+    if state == 'present' and module.params.get("status", "enabled") == "enabled":  # allow deleting/disabling a rule by id/prefix
```

Comment:
if the user hasn't provided `status` module.params.get('status') should be == "enabled" since it's defaulting to that in the arg spec. You should be able to just do `if state == 'present' and module.params["status"] == "enabled":`